transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/FragmentationIT.java

```diff
@@ -0,0 +1,394 @@
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+/**
+ * RFC-6455, section 5.4 "Fragmentation"
+ */
+public class FragmentationIT {
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/fragmentation");
+    
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/echo"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("http://localhost:8001")
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+    
+    private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(gateway).around(timeout);
+
+    @Test
+    @Specification({
+        "client.send.continuation.payload.length.125.not.fragmented/handshake.request.and.frame"
+        })
+    public void shouldFailWebSocketConnectionWhenClientSendContinuationFrameWithPayloadNotFragmented() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.continuation.payload.length.125.fragmented/handshake.request.and.frames"
+        })
+    public void shouldFailWebSocketConnectionWhenClientSendContinuationFrameWithPayloadFragmented() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.echo.text.payload.length.125.not.fragmented/handshake.request.and.frame"
+        })
+    public void shouldEchoClientSendTextFrameWithPayloadNotFragmented() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Gateway disconnected instead of echoing code, client did not read")
+    @Specification({
+        "client.echo.text.payload.length.0.fragmented/handshake.request.and.frames"
+        })
+    public void shouldEchoClientSendTextFrameWithEmptyPayloadFragmented() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Gateway writing wrong bytes back to client, read does not match")
+    @Specification({
+        "client.echo.text.payload.length.0.fragmented.with.injected.ping.pong/handshake.request.and.frames"
+        })
+    public void shouldEchoClientSendTextFrameWithEmptyPayloadFragmentedAndInjectedPingPong() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Incorrect data write by Gateway, not echoing")
+    @Specification({
```

Comment:
Gateway just sends data in multiple fragments. Spec test provides one of the ways data is received but gateway(or any implementation) may not echo the data in the spec expected framing. Spec needs to provide different scripts(or some sort of websocket framing knowledge should be built in robot).